app/models/game.rb

```diff
@@ -1,11 +1,61 @@
 class Game < ActiveRecord::Base
-	has_many :users
+
+	has_one :white_player, class_name: :user
+	has_one :black_player, class_name: :user
+	
+	has_many :pieces
+	# belongs_to :type
+
+	after_create :populate_board!
 
 	scope :available, -> {
```

Comment:
For `available`

Now that we have has_one white_player & black_player. We can simplify it. :smile:

This will work

`scope :available, -> { where('black_player_id IS NULL OR white_player_id IS NULL' ) }`